test/vfd_swmr_group_writer.c

```diff
@@ -2981,58 +3304,2076 @@ verify_group(state_t *s, unsigned int which)
+        printf("verify the compact link storage failed for 'link compact to dense' test\n");
+        goto error;
+    }
+
+    if (vrfy_close_group_id(s, g) == false) {
+        H5_FAILED();
+        AT();
+        printf("verify the group close for 'link compact to dense' test\n");
+        goto error;
+    }
+    return true;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Gclose(g);
+    }
+    H5E_END_TRY;
+    return false;
+}
+
+/*-------------------------------------------------------------------------
+ * Function:    verify_group_operations
+ *
+ * Purpose:     verify the success of group creation and
+ *              carry out the test for attribute operations(add,delete etc.)
+ *              according to the attribute test pattern.
+ *
+ * Parameters:  state_t *s
+ *              The struct that stores information of HDF5 file, named pipe
+ *              and some VFD SWMR configuration parameters
+ *
+ *              unsigned int which
+ *              The number of iterations for group creation
+ *
+ *
+ * Return:      Success:    true
+ *              Failure:    false
+ *
+ * Note:        This is called by the main() function.
+ *-------------------------------------------------------------------------
+ */
+
+static bool
+verify_group_operations(state_t *s, unsigned int which)
+{
+    bool ret_value    = false;
+    char test_pattern = s->grp_op_pattern;
+
+    switch (test_pattern) {
+        case 'c':
+            ret_value = vrfy_create_group(s, which);
+            break;
+        case 'd':
+            ret_value = vrfy_delete_group(s, which);
+            break;
+        case 'm':
+            ret_value = vrfy_move_group(s, which);
+            break;
+        case 'i':
+            ret_value = vrfy_insert_links(s, which);
+            break;
+        case 'D':
+            ret_value = vrfy_delete_links(s, which);
+            break;
+        case 't':
+            ret_value = vrfy_transit_storage_compact_to_dense(s, which);
+            break;
+        case 'T':
+            ret_value = vrfy_transit_storage_dense_to_compact(s, which);
+            break;
+        case ' ':
+        default:
+            ret_value = verify_group(s, which);
+            break;
+    }
+    return ret_value;
+}
+
+int
+main(int argc, char **argv)
+{
+    hid_t                 fapl = H5I_INVALID_HID, fcpl = H5I_INVALID_HID;
+    unsigned              step;
+    bool                  writer = false;
+    state_t               s;
+    const char *          personality;
+    H5F_vfd_swmr_config_t config;
+    const char *          fifo_writer_to_reader = "./fifo_group_writer_to_reader";
+    const char *          fifo_reader_to_writer = "./fifo_group_reader_to_writer";
+    int                   fd_writer_to_reader = -1, fd_reader_to_writer = -1;
+    int                   notify = 0, verify = 0;
+    bool                  wg_ret = false;
+    bool                  vg_ret = false;
+
+    if (!state_init(&s, argc, argv)) {
+        H5_FAILED();
```

Comment:
I suggest to replace all the occurrences of the following to "TEST_ERROR"
[ H5_FAILED(); AT(); goto error;]

TEST_ERROR is a macro defined in h5test.h and is generally used by the tests in the test directory.  (See other tests in the test directory.)

Reply:
Good, I will replace "[ H5_FAILED(); AT(); goto error;]" with TEST_ERROR. The " goto error2" has to stay the same.

Reply:
Updated the code.